src/network/heading_message.cpp

```diff
@@ -80,20 +102,24 @@ void HeadingMessageHandlerSubSystem::Execute( Message const& message )
 {
     HeadingMessage const& msg = static_cast<HeadingMessage const&>( message );
 
-    Opt<Actor> actor = mScene.GetActor( msg.mActorGUID );
-    if ( !actor.IsValid() )
+    if ( mProgramState.mMode == ProgramState::Server
+         || ( mProgramState.mMode == ProgramState::Client && msg.mActorGUID != mProgramState.mControlledActorGUID ) )
     {
-        L1( "cannot find actor with GUID: (that is not possible) %d \n", msg.mActorGUID );
-        return;
-    }
+        Opt<Actor> actor = mScene.GetActor( msg.mActorGUID );
+        if ( !actor.IsValid() )
+        {
+            L1( "cannot find actor with GUID: (that is not possible) %d \n", msg.mActorGUID );
+            return;
+        }
 
-    Opt<IMoveComponent> moveC = actor->Get<IMoveComponent>();
-    if ( !moveC.IsValid() )
-    {
-        L1( "heading is called on an actor that has no heading_component \n" );
-        return;
+        Opt<IMoveComponent> moveC = actor->Get<IMoveComponent>();
+        if ( !moveC.IsValid() )
+        {
+            L1( "heading is called on an actor that has no heading_component \n" );
```

Comment:
the log is a lie. well, not exactly, it just states the obvious, as there's no such thing as heading_component